### PR TITLE
Handle x extension properties

### LINKED ISSUE #100: Should x-* references be parsed?
I'm using x-* extensions in a number of places and wanted to split up my OAS doc over multiple files.  If I have a `x-subscription` content inline in the main file at root level, like this:

```
    "x-subscriptions": [
        {
            "on": "some.event",
            "do": [
                {
                    //...
                }
            ]
        }
```

Then after I do `$specs = Reader::readFromJsonFile(realpath($specPath));` I can access the subscriptions, eg `var_dump($spec->{'x-subscriptions'});` gives me:

```
array(1) {
  [0]=>
  array(2) {
    ["on"]=>
    string(32) "some.event"
    ["do"]=>
    array(1) {
        ...
```

But if I have:

```
    "x-subscriptions": {
        "$ref": "./x-subscriptions/some-event.json"
    },
```

with the full array moved to another file, the same `$spec` calls returns:

```
array(1) {
  ["$ref"]=>
  string(46) "./x-subscriptions/some-event.json"
}
```

I've tried a variety of ways (using 'allOf', having the array be in the root and reference individual events, etc.) but the reference doesn't resolve.

So I guess my question is; should it?

Thanks!

### PR DESCRIPTION
**Pull request by marcelthole**
This relates to #111 and fixes #100

### CODE REVIEW

File: src/SpecBaseObject.php [hunk 1]
```diff
@@ -484,4 +484,21 @@ public function getDocumentPosition(): ?JsonPointer
...
+    /**
+     * Handle extension properties with `x-` prefix.
+     * See https://github.com/OAI/OpenAPI-Specification/blob/3.0.2/versions/3.0.2.md#specificationExtensions
+     * @return array<string, mixed>
+     */
+    public function getExtensions(): array
+    {
+        $extensions = [];
+        foreach ($this->_properties as $propertyKey => $extension) {
+            if (mb_strpos($propertyKey, 'x-') !== 0) {
```
Comment (cebe) [inline src/SpecBaseObject.php:497]: why `mb_strpos` and not `strpos`? should be the same on characters `x-` as far as I see.

File: src/SpecBaseObject.php [hunk 2]
```diff
@@ -484,4 +484,21 @@ public function getDocumentPosition(): ?JsonPointer
     {
         return $this->_jsonPointer;
     }
+
+    /**
+     * Handle extension properties with `x-` prefix.
+     * See https://github.com/OAI/OpenAPI-Specification/blob/3.0.2/versions/3.0.2.md#specificationExtensions
```
Comment (cebe) [inline src/SpecBaseObject.php:490]:
```suggestion
     * Returns extension properties with `x-` prefix.
     * @see https://github.com/OAI/OpenAPI-Specification/blob/3.0.2/versions/3.0.2.md#specificationExtensions
```

File: src/SpecBaseObject.php (hunk 1 above)
Comment (marcelthole) [reply src/SpecBaseObject.php:497]: it's a habit. I always use `mb_` functions in all kind of string searches. But yeah, i don't think it's really required in this case.

Comment (cebe) [reply src/SpecBaseObject.php:497]:
```suggestion
            if (strpos($propertyKey, 'x-') !== 0) {
```

File: src/SpecBaseObject.php [hunk 3]
```diff
@@ -484,4 +484,21 @@ public function getDocumentPosition(): ?JsonPointer
     {
         return $this->_jsonPointer;
     }
+
+    /**
+     * Returns extension properties with `x-` prefix.
+     * @see https://github.com/OAI/OpenAPI-Specification/blob/3.0.2/versions/3.0.2.md#specificationExtensions
+     * @return array<string, mixed>
+     */
```
Comment (cebe) [inline src/SpecBaseObject.php:492]:
```suggestion
     * @since 1.5.3
     */
```

**Comment by cebe** [comment]
Thank you!

**Pull request merged**